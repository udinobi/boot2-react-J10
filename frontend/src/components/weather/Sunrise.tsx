import React from "react"

import styled from "styled-components"

import { SmallerText } from "../lib/StyledText"

export interface Data {
    sunrise: string
}

const Sunrise = styled(SmallerText)`
    display: inline;
    font-family: "Roboto", "sans-serif";
`

const Icon = styled.i`
    color: #EE671E;
    font-size: 1.4rem;
    padding-right: 8px;
    margin-top: 20px;
`

export default (props: Data) => 
    <div>
        <Icon className="wi wi-sunrise" />
        <Sunrise>{props.sunrise}</Sunrise>
    </div>
