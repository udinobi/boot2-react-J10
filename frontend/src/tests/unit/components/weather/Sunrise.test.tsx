import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import React from "react"

import Sunrise, { Data } from "../../../../components/weather/Sunrise"

describe("--> Snapshot for Components > weather > 'Sunrise'", () => {
    let props: Data

    beforeEach(() => {
        props = {
            sunrise: ""
        }
    })

    it("matches", () => {
        const wrapper = shallow(<Sunrise {...props}/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})
