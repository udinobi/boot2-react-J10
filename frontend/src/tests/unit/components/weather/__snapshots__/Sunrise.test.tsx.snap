// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Snapshot for Components > weather > 'Sunrise' matches 1`] = `
<div>
  <styled.i
    className="wi wi-sunrise"
  />
  <Styled(Styled(Styled(styled.div))) />
</div>
`;
